### PR TITLE
Travis merge base

### PR DESCRIPTION
**Pull request by daschuer**
This should fix some flaky CI tests on Travis

### CODE REVIEW

File: .travis.yml
Comment (Holzhaus) [inline .travis.yml:53]:
```suggestion
        - pre-commit run --from-ref $TRAVIS_BRANCH --to-ref HEAD --show-diff-on-failure
```

**Review by Holzhaus — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml
Comment (Holzhaus) [inline .pre-commit-config.yaml:58]: c/cpp/h are already taken care of by clang-format and JS files are fixed via eslint. But why exclude the other types?

Comment (daschuer) [reply .pre-commit-config.yaml:58]: I have just copied the this line from the clang-format hook without much thinking. 
```
files: \.(c|cc|cxx|cpp|frag|glsl|h|hpp|hxx|ih|ispc|ipp|java|js|m|mm|proto|vert)$
```
Do we need to adjust both then?

**Comment by daschuer** [comment]
All green now :-)

File: .travis.yml
Comment (Holzhaus) [inline .travis.yml:53]: Won't this just compare the current with the previous commit?

From https://pre-commit.com/#usage-in-continuous-integration:
> To check only files which have changed, which may be faster, use something like `pre-commit run --from-ref origin.HEAD --to-ref HEAD`.

Comment (daschuer) [reply .travis.yml:53]: Travis is checking out the PR commit. That is a virtual merge commit that merges the PR branch with the merge base. The PR Branch is HEAD^2 and the merge base at HEAD^1. 
pre-commit uses diff HEAD^1...HEAD, which crawls automatically up. 
The only "issue" is that Travis did not retest the actual merge, before pressing the merge button if the HEAD^1 as advanced. I will add a comment.

Comment (daschuer) [reply .travis.yml:55]: origin/HEAD is the head of the default branch (master in our case). That's why we have false positives.

**Comment by daschuer** [comment]
Done.

**Comment by uklotzde** [comment]
@Holzhaus Merge?

**Comment by Holzhaus** [comment]
Yes. Thanks for taking care.

**Pull request merged**
